Tests/SWBTaskConstructionTests/SwiftTaskConstructionTests.swift

```diff
@@ -3917,6 +3917,88 @@ fileprivate struct SwiftTaskConstructionTests: CoreBasedTests {
             results.checkNoDiagnostics()
         }
     }
+
+    @Test(.requireSDKs(.host))
+    func warningAsWarningsGroups() async throws {
+        try await withTemporaryDirectory { tmpDir in
+            let testProject = try await TestProject(
+                "ProjectName",
+                sourceRoot: tmpDir.join("srcroot"),
+                groupTree: TestGroup(
+                    "SomeFiles",
+                    children: [
+                        TestFile("File1.swift"),
+                    ]),
+                targets: [
+                    TestStandardTarget(
+                        "TargetName",
+                        type: .framework,
+                        buildConfigurations: [
+                            TestBuildConfiguration("Debug", buildSettings: [
+                                "SWIFT_WARNINGS_AS_WARNINGS_GROUPS": "-Wwarning",
```

Comment:
Instead of the flag itself, the argument to this setting should be something like `DeprecatedDeclaration` or `Unsafe` as a way of controlling the behavior of the diagnostic groups defined here: https://github.com/swiftlang/swift/blob/main/include/swift/AST/DiagnosticGroups.def. It's important that it's a StringList typed setting so that clients can specify an arbitrary list of groups

Reply:
This makes a lot of sense. Thanks for the clarification! 

Just to make sure, that these diagnostic groups would be same for both the new settings?